File: 카카오(프로그래머스)/lv1/개인정보수집하기.py
def solution(today, terms, privacies):
    answer = []
    termMap = dict()
    
    for term in terms:
        t, valid = term.split()
        termMap[t] = int(valid) * 28
    
    t_year, t_month, t_day = list(map(int, today.split('.')))
    
    for i in range(len(privacies)):
        p = privacies[i]
        date, term = p.split()
        year, month, day = list(map(int, date.split('.')))
        
        if (t_year-year) * 12 * 28 + (t_month-month) * 28 + (t_day-day) >= termMap[term]:
            answer.append(i+1)
    
    return answer

File: 카카오(프로그래머스)/lv1/개인정보수집하기.kt
class Solution {
    fun solution(today: String, terms: Array<String>, privacies: Array<String>): IntArray {
        var answer: MutableList<Int> = mutableListOf()
        var termMap: MutableMap<String, Int> = mutableMapOf()
        
        for (term in terms) {
            val (t, valid) = term.split(" ")
            termMap.put(t, valid.toInt()*28)
        }
        
        val (t_year, t_month, t_day) = today.split(".").map { it.toInt() }
        
        for (i in 0 until privacies.size) {
            val p = privacies[i]
            val (date, term) = p.split(" ")
            val (year, month, day) = date.split(".").map { it.toInt() }
            
            termMap[term]?.let {
                if ( (t_year-year)*12*28 + (t_month-month)*28 + (t_day-day) >= it ) {
                    answer.add(i+1)
                }
            }
        }
        
        return answer.toIntArray()
    }
}